### PR TITLE
[vcpkg baseline][sdformat13] Fix find dependency urdfdom

### PR DESCRIPTION
**Pull request by JackBoosY**
Fix configure error:
```
-- Checking for module 'urdfdom'
--   Package 'console_bridge', required by 'urdfdom', not found
CMake Error at D:/downloads/tools/cmake-3.27.1-windows/cmake-3.27.1-windows-i386/share/cmake-3.27/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find GzURDFDOM (missing: GzURDFDOM_FOUND)
```

Related: https://github.com/microsoft/vcpkg/pull/33330/

### CODE REVIEW

File: ports/sdformat13/fix-find-urdfdom.patch [hunk 1]
```diff
@@ -0,0 +1,24 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index e797e8a..0f652d7 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -56,7 +56,18 @@ if (BUILD_SDF)
+   #  2. if USE_INTERNAL_URDF is set to True, use the internal copy
+   #  3. if USE_INTERNAL_URDF is set to False, force to search system installation, fail on error
+   if (NOT DEFINED USE_INTERNAL_URDF OR NOT USE_INTERNAL_URDF)
+-    gz_find_package(GzURDFDOM)
++    find_package(urdfdom CONFIG REQUIRED)
```
Comment (JackBoosY) [inline ports/sdformat13/fix-find-urdfdom.patch:10]: REF: https://github.com/gazebosim/gz-cmake/blob/ec97ec2bd0fad0ba36704832a15e68d1781c6c61/cmake/FindGzURDFDOM.cmake#L39

**Comment by MonicaLiu0311** [comment]
Will resolve CI Pipelines failures: https://dev.azure.com/vcpkg/public/_build/results?buildId=93404&view=results

**Comment by MonicaLiu0311** [comment]
usage:
```
sdformat13 provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(sdformat13 CONFIG REQUIRED)
    target_link_libraries(main PRIVATE sdformat13::core sdformat13::requested sdformat13::sdformat13 sdformat13::using_parser_urdf)
```

When testing usage, the following error occurs:
```
1> [CMake] -- Configuring done (10.0s)
1> [CMake] CMake Error at G:/sdformat13/installed/x64-windows/share/sdformat13/sdformat13-targets.cmake:60 (set_target_properties):
1> [CMake]   The link interface of target "sdformat13::using_parser_urdf" contains:
1> [CMake] 
1> [CMake]     GzURDFDOM::GzURDFDOM
1> [CMake] 
1> [CMake]   but the target was not found.  Possible reasons include:
1> [CMake] 
1> [CMake]     * There is a typo in the target name.
1> [CMake]     * A find_package call is missing for an IMPORTED target.
1> [CMake]     * An ALIAS target is missing.
1> [CMake] 
1> [CMake] Call Stack (most recent call first):
1> [CMake]   G:/sdformat13/installed/x64-windows/share/sdformat13/sdformat13-config.cmake:124 (include)
1> [CMake]   G:/sdformat13/scripts/buildsystems/vcpkg.cmake:859 (_find_package)
1> [CMake]   CMakeUsage/CMakeLists.txt:13 (find_package)
1> [CMake] -- Generating done (0.0s)
```

<details>
<summary><b>CMakeFindUsage.cpp</b></summary>
<pre>
#include &#60;iostream&#62;
#include "sdformat.hh"

using namespace std;

int main()
{
        cout << "Hello CMake." << endl;
        return 0;
}
</pre></details>


<details>
<summary><b>CMakeLists.txt</b></summary>
<pre>
cmake_minimum_required (VERSION 3.8)

set(CMAKE_TOOLCHAIN_FILE "E:/sdformat13/scripts/buildsystems/vcpkg.cmake")

project ("CMakeFindUsage")

add_library (main "CMakeFindUsage.cpp")

find_package(sdformat13 CONFIG REQUIRED)
target_link_libraries(main PRIVATE sdformat13::core sdformat13::requested sdformat13::sdformat13 sdformat13::using_parser_urdf)
</pre></details>

**Comment by JackBoosY** [comment]
We also need to fix the config file.

File: ports/sdformat13/portfile.cmake [hunk 2]
```diff
@@ -20,14 +21,26 @@ vcpkg_cmake_configure(
...
+vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/share/${PORT}/sdformat13-config.cmake" "find_package(TINYXML2" [[
+find_package(urdfdom CONFIG ${gz_package_quiet} ${gz_package_required})
+add_library(GzURDFDOM::GzURDFDOM INTERFACE IMPORTED)
+target_link_libraries(GzURDFDOM::GzURDFDOM
+  INTERFACE
+    urdfdom::urdfdom_model
+    urdfdom::urdfdom_world
+    urdfdom::urdfdom_sensor
+    urdfdom::urdfdom_model_state
+)
```
Comment (JackBoosY) [inline ports/sdformat13/portfile.cmake:41]:
```suggestion
if (NOT TARGET GzURDFDOM::GzURDFDOM)
    find_package(urdfdom CONFIG ${gz_package_quiet} ${gz_package_required})
    add_library(GzURDFDOM::GzURDFDOM INTERFACE IMPORTED)
    target_link_libraries(GzURDFDOM::GzURDFDOM
        INTERFACE
        urdfdom::urdfdom_model
        urdfdom::urdfdom_world
        urdfdom::urdfdom_sensor
        urdfdom::urdfdom_model_state
    )
endif()
```

**Review by JackBoosY — COMMENTED** [review]
maybe better.

**Comment by MonicaLiu0311** [comment]
The usage test passed on `x64-windows` (header files found):
```
sdformat13 provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(sdformat13 CONFIG REQUIRED)
    # note: 1 additional targets are not displayed.
    target_link_libraries(main PRIVATE sdformat13::core GzURDFDOM::GzURDFDOM sdformat13::requested sdformat13::sdformat13)
```

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**